src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,30 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+            return country;
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can not add a country to DB! ", e);
```

Comment:
also, fix everywhere
```suggestion
            throw new DataProcessingException("Can not add a country to DB: " + country, e);
```